pkgs/development/python-modules/cylp/default.nix

```
@@ -0,0 +1,73 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, numpy
+, scipy
+, python
+, pytestCheckHook
+, cython
+, cbc
+, bzip2
+, zlib
+, pkg-config
+}:
+
+buildPythonPackage rec {
+  pname = "cylp";
+  version = "0.91.4";
+
+  src = fetchFromGitHub {
+    owner = "coin-or";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-m09s+QrkLGtF83tDLNjU4ECOqgW5RT0h9+fSwDeADag=";
+  };
+
+  nativeBuildInputs = [
+    pkg-config
+    cython
+  ];
+
+  buildInputs = [
+    cbc
+    bzip2
+    zlib
+  ];
+
+  propagatedBuildInputs = [
+    numpy
+    scipy
+  ];
+
+  preBuild = ''
+    # Leak some memory, reverting this commit. Otherwise there's a segfault
+    # that's being debugged in https://github.com/coin-or/CyLP/issues/105
+    substituteInPlace cylp/cy/CyClpSimplex.pyx --replace \
+      'del self.CppSelf' \
+      'pass'
+
+    # Rebuild all cython files
+    export CYLP_USE_CYTHON=1
+  '';
+
+  checkInputs = [ pytestCheckHook ];
+  pythonImportsCheck = [ "cylp" ];
+  pytestFlagsArray = [ "-s" ];
+
+  preCheck = ''
+    cd cylp/tests
+    rm __init__.py
```

Why? pytest shouldn't care about this (either exact directory or .

If decide to leave:
```suggestion
pytestFlagsArray = [ ... "cylp/tests"];
preCheck = ''
  rm cylp/tests/__init__.py
'';
```

---

I'm not entirely sure, but pytest gets confused with the local imports, since the tests are inside the source directory, if i don't do this.

---

If the package does something funky with local imports, I've had luck with either moving the tests to a separate directory, or changing pytest's import more ``--import-mode=append``. For future reference: https://docs.pytest.org/en/6.2.x/pythonpath.html#import-modes